### PR TITLE
Adding DPUB to protocol spec and to nsqd

### PR DESCRIPTION
**Pull request by spruce**
Hi,

Adding DPUB to the docs as requested here: https://github.com/nsqio/nsq/issues/795

Let me know if something is wrong.

### CODE REVIEW

File: _posts/2012-12-01-nsqd.md [hunk 1]
```diff
@@ -137,14 +137,15 @@ can optionally listen on a third port for HTTPS.
   Query Params:
 
       topic - the topic to publish to
+      defer - for how many ms to delay message delivery
```
Comment (ploxiln) [inline _posts/2012-12-01-nsqd.md]: Please mention that defer is optional.

I might not include "defer" in the example - it's not the typical way, or the simplest way, to use "/pub".

File: _posts/2013-03-01-tcp_protocol_spec.md [hunk 2]
```diff
@@ -220,6 +220,35 @@ Error Responses:
...
+Publish a message to a **topic** deferred:
+
+NOTE: available in `nsqd` `v0.3.6+`
+
+```
+DPUB <topic_name> <defer_time>\n
+[ 4-byte size in bytes ][ N-byte binary data ]
+
+<topic_name> - a valid string (optionally having #ephemeral suffix)
+<defer_time> - a valid defer time (range 0 - maxReqTimout)
```
Comment (ploxiln) [inline _posts/2013-03-01-tcp_protocol_spec.md]: Like the other command descriptions, should probably say "a string representation of integer ..."

**Comment by ploxiln** [comment]
Thanks for patching up the docs!
Only had a couple minor quibbles.

**Comment by spruce** [comment]
Thanks for the comments. How does it look now?

File: _posts/2013-03-01-tcp_protocol_spec.md [hunk 3]
```diff
@@ -220,6 +220,35 @@ Error Responses:
     E_BAD_MESSAGE
     E_MPUB_FAILED
 
+#### DPUB
+
+Publish a message to a **topic** deferred:
```
Comment (mreiferson) [inline _posts/2013-03-01-tcp_protocol_spec.md]: > Publish a deferred message to a **topic**

File: _posts/2012-12-01-nsqd.md [hunk 4]
```diff
@@ -137,6 +137,7 @@ can optionally listen on a third port for HTTPS.
   Query Params:
 
       topic - the topic to publish to
+      defer - optional, for how many ms to delay message delivery
```
Comment (mreiferson) [inline _posts/2012-12-01-nsqd.md]: > defer - the time in ms to delay message delivery (optional)

**Comment by mreiferson** [comment]
Sorry! Two minor comments 😁

File: _posts/2012-12-01-nsqd.md [hunk 5]
```diff
@@ -1,4 +1,4 @@
----
+F---
```
Comment (mreiferson) [inline _posts/2012-12-01-nsqd.md]: think this change was unintentional?

**Comment by mreiferson** [comment]
I was able to edit these directly and land this, thanks!!!

File: _posts/2012-12-01-nsqd.md (hunk 5 above)
Comment (spruce) [inline _posts/2012-12-01-nsqd.md]: yes :100:

**Pull request merged**